### ISSUE TITLE
[*] rtsx_usb depends badly on autosuspend

### ISSUE THREAD

**Issue by alondrayona** [StarLabsLtd/firmware#183]
Entering into S3 is super slow, it takes about 15-20 seconds.

At first I thought it was caused by the iwlwifi module, but even if I remove it before sleep, it still takes a long time.

This happens both on a Fedora 40 live ISO and on an up-to-date installation (Linux 6.9.7-gentoo-dist / linux-firmware 20240610 / firmware 2024.06) 

<details>
<summary>dmesg</summary>

```
[ 5563.904243] elogind-daemon[1740]: Suspending system...
[ 5563.904261] PM: suspend entry (deep)
[ 5563.907497] Filesystems sync: 0.003 seconds
[ 5563.907723] Loading firmware: iwlwifi-so-a0-jf-b0-77.ucode
[ 5563.907756] Loading firmware: regulatory.db
[ 5563.907797] Loading firmware: i915/adlp_dmc.bin
[ 5563.907858] Loading firmware: i915/tgl_guc_70.bin
[ 5563.907993] Loading firmware: i915/tgl_huc.bin
[ 5563.908468] Loading firmware: intel/ibt-0040-2120.sfi
[ 5563.909032] Loading firmware: intel/ibt-0040-2120.ddc
[ 5563.909081] Loading firmware: regulatory.db.p7s
[ 5564.026340] Freezing user space processes
[ 5564.028631] Freezing user space processes completed (elapsed 0.002 seconds)
[ 5564.028638] OOM killer disabled.
[ 5564.028640] Freezing remaining freezable tasks
[ 5580.829045] Freezing remaining freezable tasks completed (elapsed 16.800 seconds)
[ 5580.829088] printk: Suspending console(s) (use no_console_suspend to debug)
[ 5581.066959] iwlwifi 0000:00:14.3: timeout waiting for FW reset ACK (inta_hw=0x0)
[ 5581.067106] iwlwifi 0000:00:14.3: Start IWL Error Log Dump:
[ 5581.067109] iwlwifi 0000:00:14.3: Transport status: 0x0000004A, valid: 6
[ 5581.067115] iwlwifi 0000:00:14.3: Loaded firmware version: 77.f92b5fed.0 so-a0-jf-b0-77.ucode
[ 5581.067120] iwlwifi 0000:00:14.3: 0x00000084 | NMI_INTERRUPT_UNKNOWN       
[ 5581.067125] iwlwifi 0000:00:14.3: 0x000002F0 | trm_hw_status0
[ 5581.067128] iwlwifi 0000:00:14.3: 0x00000000 | trm_hw_status1
[ 5581.067131] iwlwifi 0000:00:14.3: 0x004D2A06 | branchlink2
[ 5581.067134] iwlwifi 0000:00:14.3: 0x004C92BE | interruptlink1
[ 5581.067137] iwlwifi 0000:00:14.3: 0x004C92BE | interruptlink2
[ 5581.067139] iwlwifi 0000:00:14.3: 0x00011B6A | data1
[ 5581.067142] iwlwifi 0000:00:14.3: 0x01000000 | data2
[ 5581.067145] iwlwifi 0000:00:14.3: 0x00000000 | data3
[ 5581.067148] iwlwifi 0000:00:14.3: 0x00000000 | beacon time
[ 5581.067150] iwlwifi 0000:00:14.3: 0x08A496EA | tsf low
[ 5581.067153] iwlwifi 0000:00:14.3: 0x00000000 | tsf hi
[ 5581.067155] iwlwifi 0000:00:14.3: 0x00000000 | time gp1
[ 5581.067158] iwlwifi 0000:00:14.3: 0x08A557AE | time gp2
[ 5581.067160] iwlwifi 0000:00:14.3: 0x00000001 | uCode revision type
[ 5581.067163] iwlwifi 0000:00:14.3: 0x0000004D | uCode version major
[ 5581.067166] iwlwifi 0000:00:14.3: 0xF92B5FED | uCode version minor
[ 5581.067169] iwlwifi 0000:00:14.3: 0x00000370 | hw version
[ 5581.067172] iwlwifi 0000:00:14.3: 0x00480002 | board version
[ 5581.067174] iwlwifi 0000:00:14.3: 0x8025FF00 | hcmd
[ 5581.067177] iwlwifi 0000:00:14.3: 0x00020000 | isr0
[ 5581.067179] iwlwifi 0000:00:14.3: 0x00000000 | isr1
[ 5581.067182] iwlwifi 0000:00:14.3: 0x48F04802 | isr2
[ 5581.067184] iwlwifi 0000:00:14.3: 0x00C3400C | isr3
[ 5581.067187] iwlwifi 0000:00:14.3: 0x00000000 | isr4
[ 5581.067189] iwlwifi 0000:00:14.3: 0x00DB0148 | last cmd Id
[ 5581.067192] iwlwifi 0000:00:14.3: 0x00011B6A | wait_event
[ 5581.067194] iwlwifi 0000:00:14.3: 0x00000000 | l2p_control
[ 5581.067197] iwlwifi 0000:00:14.3: 0x00000000 | l2p_duration
[ 5581.067199] iwlwifi 0000:00:14.3: 0x00000000 | l2p_mhvalid
[ 5581.067202] iwlwifi 0000:00:14.3: 0x00000000 | l2p_addr_match
[ 5581.067204] iwlwifi 0000:00:14.3: 0x00000018 | lmpm_pmg_sel
[ 5581.067207] iwlwifi 0000:00:14.3: 0x00000000 | timestamp
[ 5581.067210] iwlwifi 0000:00:14.3: 0x0000F0EC | flow_handler
[ 5581.067292] iwlwifi 0000:00:14.3: 0x00000000 | ADVANCED_SYSASSERT
[ 5581.067295] iwlwifi 0000:00:14.3: 0x00000000 | umac branchlink1
[ 5581.067298] iwlwifi 0000:00:14.3: 0x00000000 | umac branchlink2
[ 5581.067300] iwlwifi 0000:00:14.3: 0x00000000 | umac interruptlink1
[ 5581.067303] iwlwifi 0000:00:14.3: 0x00000000 | umac interruptlink2
[ 5581.067305] iwlwifi 0000:00:14.3: 0x00000000 | umac data1
[ 5581.067307] iwlwifi 0000:00:14.3: 0x00000000 | umac data2
[ 5581.067310] iwlwifi 0000:00:14.3: 0x00000000 | umac data3
[ 5581.067312] iwlwifi 0000:00:14.3: 0x00000000 | umac major
[ 5581.067314] iwlwifi 0000:00:14.3: 0x00000000 | umac minor
[ 5581.067317] iwlwifi 0000:00:14.3: 0x00000000 | frame pointer
[ 5581.067319] iwlwifi 0000:00:14.3: 0x00000000 | stack pointer
[ 5581.067322] iwlwifi 0000:00:14.3: 0x00000000 | last host cmd
[ 5581.067324] iwlwifi 0000:00:14.3: 0x00000000 | isr status reg
[ 5581.067383] iwlwifi 0000:00:14.3: IML/ROM dump:
[ 5581.067385] iwlwifi 0000:00:14.3: 0x00000B03 | IML/ROM error/state
[ 5581.067443] iwlwifi 0000:00:14.3: 0x000050CE | IML/ROM data1
[ 5581.067504] iwlwifi 0000:00:14.3: 0x00000090 | IML/ROM WFPM_AUTH_KEY_0
[ 5581.067560] iwlwifi 0000:00:14.3: Fseq Registers:
[ 5581.067611] iwlwifi 0000:00:14.3: 0x60000000 | FSEQ_ERROR_CODE
[ 5581.067666] iwlwifi 0000:00:14.3: 0x00330000 | FSEQ_TOP_INIT_VERSION
[ 5581.067718] iwlwifi 0000:00:14.3: 0x00010009 | FSEQ_CNVIO_INIT_VERSION
[ 5581.067770] iwlwifi 0000:00:14.3: 0x0000A384 | FSEQ_OTP_VERSION
[ 5581.067825] iwlwifi 0000:00:14.3: 0xC75441EA | FSEQ_TOP_CONTENT_VERSION
[ 5581.067877] iwlwifi 0000:00:14.3: 0x4552414E | FSEQ_ALIVE_TOKEN
[ 5581.067927] iwlwifi 0000:00:14.3: 0x00080400 | FSEQ_CNVI_ID
[ 5581.067977] iwlwifi 0000:00:14.3: 0x01300202 | FSEQ_CNVR_ID
[ 5581.068031] iwlwifi 0000:00:14.3: 0x00080400 | CNVI_AUX_MISC_CHIP
[ 5581.068084] iwlwifi 0000:00:14.3: 0x01300202 | CNVR_AUX_MISC_CHIP
[ 5581.068134] iwlwifi 0000:00:14.3: 0x0000485B | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[ 5581.068238] iwlwifi 0000:00:14.3: 0x0BADCAFE | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[ 5581.068291] iwlwifi 0000:00:14.3: 0x00000000 | FSEQ_PREV_CNVIO_INIT_VERSION
[ 5581.068343] iwlwifi 0000:00:14.3: 0x00330000 | FSEQ_WIFI_FSEQ_VERSION
[ 5581.068395] iwlwifi 0000:00:14.3: 0x00330000 | FSEQ_BT_FSEQ_VERSION
[ 5581.068447] iwlwifi 0000:00:14.3: 0x00000122 | FSEQ_CLASS_TP_VERSION
[ 5581.068505] iwlwifi 0000:00:14.3: UMAC CURRENT PC: 0x8047e450
[ 5581.068558] iwlwifi 0000:00:14.3: LMAC1 CURRENT PC: 0xd0
[ 5581.068611] iwlwifi 0000:00:14.3: WRT: Collecting data: ini trigger 4 fired (delay=0ms).
[ 5581.550107] PM: suspend devices took 0.724 seconds
[ 5581.600504] ACPI: EC: interrupt blocked
[ 5581.644632] ACPI: PM: Preparing to enter system sleep state S3
[ 5581.690385] ACPI: EC: event blocked
[ 5581.690388] ACPI: EC: EC stopped
[ 5581.690389] ACPI: PM: Saving platform NVS memory
[ 5581.690461] Disabling non-boot CPUs ...
[ 5581.692370] smpboot: CPU 1 is now offline
[ 5581.695551] smpboot: CPU 2 is now offline
[ 5581.698923] smpboot: CPU 3 is now offline
[ 5581.706893] ACPI: PM: Low-level resume complete
[ 5581.706981] ACPI: EC: EC started
[ 5581.706982] ACPI: PM: Restoring platform NVS memory
[ 5581.708256] Enabling non-boot CPUs ...
[ 5581.708287] smpboot: Booting Node 0 Processor 1 APIC 0x2
[ 5581.709336] CPU1 is up
[ 5581.709355] smpboot: Booting Node 0 Processor 2 APIC 0x4
[ 5581.710481] CPU2 is up
[ 5581.710496] smpboot: Booting Node 0 Processor 3 APIC 0x6
[ 5581.711645] CPU3 is up
[ 5581.712018] ACPI: PM: Waking up from system sleep state S3
[ 5581.791701] ACPI: EC: interrupt unblocked
[ 5581.818511] ACPI: EC: event unblocked
[ 5581.822685] i915 0000:00:02.0: [drm] GT0: GuC firmware i915/tgl_guc_70.bin version 70.20.0
[ 5581.822695] i915 0000:00:02.0: [drm] GT0: HuC firmware i915/tgl_huc.bin version 7.9.3
[ 5581.826005] iwlwifi 0000:00:14.3: WRT: Invalid buffer destination
[ 5581.827425] i915 0000:00:02.0: [drm] GT0: HuC: authenticated for all workloads
[ 5581.828366] i915 0000:00:02.0: [drm] GT0: GUC: submission enabled
[ 5581.828368] i915 0000:00:02.0: [drm] GT0: GUC: SLPC enabled
[ 5581.828706] i915 0000:00:02.0: [drm] GT0: GUC: RC enabled
[ 5581.837390] nvme nvme0: 4/0/0 default/read/poll queues
[ 5581.842302] nvme nvme0: Ignoring bogus Namespace Identifiers
[ 5581.928147] iwlwifi 0000:00:14.3: WFPM_UMAC_PD_NOTIFICATION: 0x1f
[ 5581.928207] iwlwifi 0000:00:14.3: WFPM_LMAC2_PD_NOTIFICATION: 0x1f
[ 5581.928269] iwlwifi 0000:00:14.3: WFPM_AUTH_KEY_0: 0x90
[ 5581.928282] iwlwifi 0000:00:14.3: CNVI_SCU_SEQ_DATA_DW9: 0x0
[ 5581.929689] iwlwifi 0000:00:14.3: RFIm is deactivated, reason = 4
[ 5582.077448] usb 1-10: reset full-speed USB device number 3 using xhci_hcd
[ 5582.219855] PM: resume devices took 0.400 seconds
[ 5582.220427] OOM killer enabled.
[ 5582.220430] Restarting tasks ... 
[ 5582.220916] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i9xx_always_on_power_well_ops [i915])
[ 5582.221818] mei_pxp 0000:00:16.0-fbf6fcf1-96cf-4e2e-a6a6-1bab8cbe36b1: bound 0000:00:02.0 (ops tfp410_ops [i915])
[ 5582.223501] done.
[ 5582.223515] random: crng reseeded on system resumption
[ 5582.409646] PM: suspend exit
[ 5582.409692] elogind-daemon[1740]: System resumed.
```

</details>

**Comment by Sean-StarLabs** [comment]
Same as #168 but your title is better

**Comment by Sean-StarLabs** [comment]
SD_CN_N is pulled low with the tray, so Linux constantly tries to scan for an SD Card. It wastes power and prevents S3 entry (it's frozen after around 13 seconds).

**Comment by edpenz** [comment]
> SD_CN_N is pulled low with the tray, so Linux constantly tries to scan for an SD Card. It wastes power and prevents S3 entry (it's frozen after around 13 seconds).

For anyone else tracking this issue, I removed my SD tray and now S3 entry only takes a second.

**Comment by rbm78bln** [comment]
> SD_CN_N is pulled low with the tray

So the tray pulls SD_CD# low constantly, right?
Looking at the realtek datasheets it should actually only be pulled low by an actual sd card, correct?

Isn't this basically a hardware design flaw (i.e. hardware bug)?

**Comment by Sean-StarLabs** [comment]
According to realtek, no. Tray should pull it low then the driver should timeout after 100ms.

**Comment by rbm78bln** [comment]
Okay cool, this is quite relaxing to know...

**Comment by AndrzejTunkiel** [comment]
Pulling tray out also fixed my unreliable Suspend issue (sometimes it does suspend, sometimes it does not).

I tried disabling the card reader in the coreboot - that did not work.

I then disabled the USB device via `echo '1-3.2' |sudo tee /sys/bus/usb/drivers/usb/unbind` and now the suspend is fast and reliable for me too.

I am on Fedora Silverblue 6.9.10-200.fc40.x86_64

**Comment by meskio** [comment]
I tried removing the sdcard tray and disabling the usb device, but my starlite never gets to sleep. When I tell it sleep it turns the screen black and keep the led on (no blinking just on all time like when the device is on). Whatever I wait doesn't seem to do anything until I force it to power down pressing the power button.

I see a similar behavior sometimes when I leave it on for a while without touching it even though I disabled the auto suspension, it should only lock, but maybe this is a bug in gnome. Same behavior also when I turn off the device, it doesn't turn off completely it stays with the screen off and the led on until I press the power button for few seconds or it drains completely the battery.

I'm using debian Trixie (testing) with kernel 6.10.6 and I just upgraded the firmware to 24.08 but the problem is still there.

In `journalctl -b -1` I don't see any more logs after `systemd-sleep[3206]: Performing sleep operation 'suspend'...`. Any other way to provide debug information to help finding the issue?

Maybe a different issue? Should I open another issue?

**Comment by Sean-StarLabs** [comment]
>  disabling the usb device

What did you do?

Definitely different,  could just be a Debian bug - are you able to test with a Live USB with Ubuntu?

**Comment by Sean-StarLabs** [comment]
Ping support - that's something else

**Comment by spiritofsands** [comment]
I have  the same issue with a long suspend switching. Workaround with unbinding the usb device from AndrzejTunkiel makes suspend instant.
I have Ubuntu 24.10, kernel 6.8.0-39-generic.
Reproduced on both 24.07 and 24.08 firmware.

**Comment by mixmastamyk** [comment]
Same, I've been choosing suspend from the menu, waiting 30 secs and then doing it again to get it to sleep.  Thanks for the work around with tee.

**Comment by Sean-StarLabs** [comment]
Workaround until upstreamed: https://github.com/StarLabsLtd/rtsx_usb

**Comment by adrianh411** [comment]
Noticed 2 things:-

1) Suspend works perfectly when the external keyboard is attached.
2) Suspend works perfectly if you have a MicroSD card in the slot.

So, tested last night with a MicroSD card in the slot, charged Starlite up to 75% and put it into Suspend overnight.
Woke up this morning and resumed perfectly and charge had only dropped to 66% over 9 hours (roughly 1% per hour).

These are still workarounds but hopefully will also provide another way to get suspend working until the software upstream is updated.

**Comment by Anachron** [comment]
I've got a bit different numbers. The sleep is `deep` but I receive around 20% battery drain in 10-12 hours time. 

For me that is a bit high. 

I'm using the `echo '1-3.2' |sudo tee /sys/bus/usb/drivers/usb/unbind` workaround.

**Comment by Sean-StarLabs** [comment]
Merged upstream https://git.kernel.org/pub/scm/linux/kernel/git/gregkh/char-misc.git/commit/?h=char-misc-next&id=235b630eda072d7e7b102ab346d6b8a2c028a772

**Comment by Anachron** [comment]
Very nice!

My `sleep battery drain` meanwhile also decreased, so that's fine on my part now too!

**Comment by mixmastamyk** [comment]
What kernel # will this land in?

**Comment by Sean-StarLabs** [comment]
6.13 I believe

**Comment by mixmastamyk** [comment]
Today I installed Fedora 41 with kernel 6.13.5, deactivated the unbind command at boot, and restarted.... 

Sleep is working promptly without the workaround!  Thanks.

**Comment by 8johnny6** [comment]
In Fedora 42 with Kernel 6.13.12 the problem reappears. (I use this Kernel because 6.14.2 and 6.14.3 won't let me use my WiFi-Card).

I've written a small script which is executed just before suspend. The script uses the Workaround acording to @Anachron. 

place this as an executable script in /usr/lib/systemd/system-sleep


```
#!/bin/bash
if [ "${1}" == "pre" ]; then
  # Do the thing you want before suspend here:
  sudo echo '1-3.2' | sudo tee /sys/bus/usb/drivers/usb/unbind
  sudo echo "unbind SD-Slot" >> /tmp/systemd_suspend_test
  sudo echo "going to Standby gewechselt at $(date)" >> /tmp/systemd_suspend_test
elif [ "${1}" == "post" ]; then
  # Do the thing you want after resume here:
  sudo echo '1-3.2' | sudo tee /sys/bus/usb/drivers/usb/bind
  sudo echo "bind SD-Slot" >> /tmp/systemd_suspend_test 
  sudo echo "Aus dem Standby aufgewacht um $(date)" >> /tmp/systemd_suspend_test
fi
```

This helped me a lot with the long time to suspend. After the buttonis pressed it takes just a few seconds to suspend.

I hope it helps someone here.  

Best regards
Johnny

**Comment by 8johnny6** [comment]
> simplest workaround is just enabling auto suspend:
> 
> ```
> echo auto | sudo tee /sys/bus/usb/devices/1-3.2/power/control
> ```

This option ist set to "auto" and i have the problems suspending. 

I'm fine with my solution, it survives a reboot, it works and with this script im also deactivating my Keyboard from waking from suspend. (i had the problem where my battery was dead, when my StarLite was in my bag, because the keyboard woke it all the time)

**Comment by mixmastamyk** [comment]
Yes, the problem came back, please reopen.  I read somewhere that the fix was reverted for causing another problem.

My systemd unit if interested:

```
⏵ cat /etc/systemd/system/fix_sleep.service
[Unit]
Description=Workaround sleep issue by deactivating SD-Card's USB port.

[Service]
ExecStart=/usr/bin/sh -c 'echo "1-3.2" | tee /sys/bus/usb/drivers/usb/unbind'

[Install]
WantedBy=multi-user.target
```
```
# alias sc='sudo systemctl'
⏵ sc enable fix_sleep
⏵ sc start fix_sleep
```

There could be a line to deactivate on stop, but I personally don't need the sd reader anytime soon.

**Comment by rbm78bln** [comment]
Hi @mixmastamyk ,

instead of using multi-user.target, you could also go for this:
```
[Unit]
...
StopWhenUnneeded=yes

[Service]
ExecStart=...
ExecStop=/usr/bin/sh -c 'echo "1-3.2" | tee /sys/bus/usb/drivers/usb/bind'

[Install]
WantedBy=sleep.target
```

or any of those as needed:

- ```suspend.target```
- ```suspend-then-hibernate.target```
- ```hybrid-sleep.target```
- ```hibernate.target```
- ```sleep.target```

In that case your sd slot would only be unbound when your devices is sent to sleep and will be rebound after waking up, very similar to what the script by @8johnny6 does.

(I must admit that I also just blacklisted that kernel module, as I simply cannot imagine a use case for an sd card reader anymore nowadays...)

**Comment by mixmastamyk** [comment]
Cool, thanks.  I didn't think of blacklisting the module, good idea.  Someone posted that command-line and I didn't look further.

Can always allow it again in several months.

**Comment by Sean-StarLabs** [comment]
> I still don't think the situation is acceptable.

Have you logged a bug with Fedora?

**Comment by mixmastamyk** [comment]
I could and don't mind helping out in general.  But, don't have any special knowledge other than "prevents sleep" and here's a link to code that supposedly fixes it.  Also, not sure if Fedora.org has the authority to make it happen.  Maybe it does, but why would they listen to a rando over a known kernel dev?

**Comment by mixmastamyk** [comment]
Well, as you can see I filed a bug and was promptly ignored.  ;-)

**Comment by Sean-StarLabs** [comment]
> Well, as you can see I filed a bug and was promptly ignored. ;-)

Can you share the link?

> BTW offtopic but, I tried to send Sean an email separately about another issue, a message with "trustworthy..." in the subject line, and would appreciate a response. Cheers,

Ping it to support@starlabs.systems - it'll reach the right place

**Comment by mixmastamyk** [comment]
It's about four comments above, the redhat link. 👆

**Comment by py0xc3** [comment]
I just became aware of this being still open. I had blacklisted `mmc_core`+`rtsx_usb_sdmmc`. That solves the issue, but obviously only for those not needing the mmc/sd card reader.

A bug ticket in Fedora is useful, but I think that is better off upstream of Fedora. I never worked with these modules before, but given that their blacklisting works, the issue might be reported against one of the two (didn't do much testing about it but just blacklisted both due to lack of time). Mixmastamyk, the bug ticket in Fedora is not ignored, but there are many bug tickets, and developers need to focus and prioritize. Keep in mind that this bug ticket at Fedora does not contain any log files. I am not sure what the maintainers should work on based on that. 

@Sean-StarLabs  have you filed the patches against github? Afaik, Linus ignores patches against github because his github repo is just a backup mirror (and for some other reasons :X ). (I didn't get into details about this but I assume it also must be not merged directly against Linus but some maintainer of a sub-sys or so?)

It might be useful to open a bug report upstream against rtsx_usb_sdmmc (I assume from context that this is more likely the origin rather than mmc_core but that's just a guess without investigation/analysis) and add or refer to the patches? I suggest to use patch links not on GitHub :) Maybe the assignee/maintainer can help to get the patches merged.

Especially when patches are attached, I could imagine that the changes are good the ticket to be processed soon. Just filing this upstream in a bug report might be easier than going through the mailing lists (?).

I didn't file anything as I have no mmc/sd cards at all. So I could not even verify if any patch break the mmc/sd to work (and I also have to prioritize :( ).

I maybe can convince our maintainers to create a Fedora kernel + Sean's patches to test it before merging upstream. But it should be already somewhere upstream (and not github:) before I would dare to ask them. So if it is in a kernel bug ticket including the patches and some confirmation from the assignee/maintainer this could be the solution, then I can reach out and see if I can get a kernel from the Fedora kernel team, and if we can confirm this works out, we might finally convince the assignee/maintainer to forward it to the mainline kernel :) It might be useful if someone here could then volunteer to verify with potential patches that they do not just solve the sleep issue but also that sd/mmc still works, just to verify (?)

If that is preferred, I may find some time in next days to file the issue upstream (assuming I can still reproduce it on my StarLite), and then you can add your patches with a post or so, if you prefer it that way?

**Comment by Sean-StarLabs** [comment]
> have you filed the patches against github? 

No, mailing list - I don't think you can in git?

> It might be useful to open a bug report upstream against rtsx_usb_sdmmc

When I pushed the first patch, there were 104 bug reports :)

I just split of a mini-patch for a detection only fix and send that up - will see if we get any response

**Comment by py0xc3** [comment]
> No, mailing list - I don't think you can in git?

Theoretically a PR can be created, but it will be ignored/rejected, as github is just used as mirror.

> When I pushed the first patch, there were 104 bug reports :)
> I just split of a mini-patch for a detection only fix and send that up - will see if we get any response

Great, thanks. If nothing comes up otherwise, I could still file a bug report against `rtsx_usb` in the next days. Then we have another channel that raises awareness at the maintainer(s). At least in some cases/subsys, the chances are high(er) with the kernel bug tracker to get the direct attention of the one (person or team) handling `rtsx_usb` in specific. I don't know how this team is organized yet, but at first glance I think there are just 4 bug tickets open that are explicitly against `rtsx_usb` rather than against other parts of MMC/SD component. 

If you provide a link to your patches in a mailing list archive (that should suffice), I would then get into details of how they want their reports filed and create a ticket with logs and link to your patches.

**Comment by Sean-StarLabs** [comment]
https://lkml.org/lkml/2026/2/19/965

**Comment by py0xc3** [comment]
It's the most imperfect report I have ever written as there are several limitations about what I can test in a representative way, but it might help to get the person/team working on `rtsx_usb`, and the patches and their elaborations should be the best indication for the issue anyway: 
https://bugzilla.kernel.org/show_bug.cgi?id=221109

-----

Just in case we reach the point that I can get us a Fedora kernel with the patches from our kernel team: is someone here who would volunteer to test that kernel about if it solves the issue with the current StarLite firmware and also if SD/MMC cards still work as expected? I can test the sleep in general and will do so, but the device at my disposal might not be firmware updated till then (currently old firmware; long story) and I have no mmc/sd cards. 

The kernel would be installable through dnf from Fedora's build system (koji) directly, so a trustworthy source for Fedora users, but as it would not be an official update that is intended for being pushed to stable but only for testing, we will not be able to sign it for secureboot. Volunteers would be nice to get this solved as reliable as possible, to finally get all that stuff closed :) 

Maybe the maintainer will just merge it, but in some cases they prefer to have stuff tested and get feedback before doing so...

**Comment by mixmastamyk** [comment]
@py0xc3 I could probably test that.

**Comment by py0xc3** [comment]
@Sean-StarLabs If I get it right, you tried a third time to send the patch upstream, in a modified variant [1] ?

I assume you also have no response of any type yet, right ? If so, I would wait a few days to see if your v2 patch is somehow processed this time, and then I would have a chat with our kernel team and see if they have an idea about this.

[1] https://patchew.org/linux/20260305194052.5120-1-sean@starlabs.systems/20260305194052.5120-2-sean@starlabs.systems/

**Comment by Sean-StarLabs** [comment]
Basically, yes - send three times, no reply, and to an old chain about the first patch - nada.

**Comment by py0xc3** [comment]
Can someone confirm if the issue still occurs on kernel 7.1.n ? I have talked to a kernel developer who suggested two things, I might try the first if the issue is still there -> I cannot test as I do not have the device always at my disposal.

**Comment by py0xc3** [comment]
@Sean-StarLabs is there a chance that there was an issue with the `CC`/`To` in your submission of the patches? I just recognized that your patches ain't in the 2025-11 history of `linux-mmc` [[1](https://lore.kernel.org/linux-mmc/?t=20251111173658)] (the maintainer of the superordinated path earlier referred me to there). Off the cuff I find them only in `linux-kernel`/LKML, but I expect that might be not correct as first instance and likely too noisy for anyone to give feedback about it.

**Comment by Sean-StarLabs** [comment]
Could be
Later ones have defo landed fine, and I'm just adjusting/testing for the latest requsts: https://github.com/StarLabsLtd/linux/tree/sean/rtsx-tray-fix-v4-20260706
